consensus/service/src/api/client_api_service.rs

```diff
@@ -83,15 +83,11 @@ impl ClientApiService {
     ) -> Result<ProposeTxResponse, ConsensusGrpcError> {
         counters::ADD_TX_INITIATED.inc();
         let tx_context = self.enclave.client_tx_propose(msg.into())?;
-        let mut response = ProposeTxResponse::new();
 
         // Cache the transaction. This performs the well-formedness checks.
         let tx_hash = self.tx_manager.insert(tx_context).map_err(|err| {
             if let TxManagerError::TransactionValidation(cause) = &err {
                 counters::TX_VALIDATION_ERROR_COUNTER.inc(&format!("{:?}", cause));
-                let result = ProposeTxResult::from(cause.clone());
-                response.set_result(result);
-                response.set_err_msg(cause.to_string());
```

Comment:
Why remove this?

Reply:
on every code path where we hit these lines, we will hit the `?` operator below (since this is part of `map_err`), so `response` will be discarded